zbench.zig

```diff
@@ -219,13 +214,69 @@ pub const Benchmark = struct {
 
-/// BenchmarkResult stores the result of a single benchmark.
-/// It includes the name and the total duration of the benchmark.
+/// BenchmarkResult stores the resulting computed metrics/statistics from a benchmark
 pub const BenchmarkResult = struct {
-    /// Name of the benchmark.
     name: []const u8,
-    /// Total duration of the benchmark in nanoseconds.
-    duration: u64,
+    percs: Percentiles,
+    avg_duration: usize,
+    std_duration: usize,
+    min_duration: usize,
+    max_duration: usize,
+    total_operations: usize,
+    total_time: usize,
+
+    /// Formats and prints the benchmark result in a readable format.
+    pub fn prettyPrint(self: BenchmarkResult, header: bool) !void {
+        var total_time_buffer: [128]u8 = undefined;
+        const total_time_str = try format.duration(total_time_buffer[0..], self.total_time);
+
+        var p75_buffer: [128]u8 = undefined;
+        const p75_str = try format.duration(p75_buffer[0..], self.percs.p75);
+
+        var p99_buffer: [128]u8 = undefined;
+        const p99_str = try format.duration(p99_buffer[0..], self.percs.p99);
+
+        var p995_buffer: [128]u8 = undefined;
+        const p995_str = try format.duration(p995_buffer[0..], self.percs.p995);
+
+        var avg_std_buffer: [128]u8 = undefined;
+        var avg_std_offset = (try format.duration(avg_std_buffer[0..], self.avg_duration)).len;
+        avg_std_offset += (try std.fmt.bufPrint(avg_std_buffer[avg_std_offset..], " ± ", .{})).len;
+        avg_std_offset += (try format.duration(avg_std_buffer[avg_std_offset..], self.std_duration)).len;
+        const avg_std_str = avg_std_buffer[0..avg_std_offset];
+
+        var min_buffer: [128]u8 = undefined;
+        const min_str = try format.duration(min_buffer[0..], self.min_duration);
+
+        var max_buffer: [128]u8 = undefined;
+        const max_str = try format.duration(max_buffer[0..], self.max_duration);
+
+        var min_max_buffer: [128]u8 = undefined;
+        const min_max_str = try std.fmt.bufPrint(min_max_buffer[0..], "({s} ... {s})", .{ min_str, max_str });
+
+        if (header) try prettyPrintHeader();
+
+        const stdout = std.io.getStdOut().writer();
+        try stdout.print(
+            "{s:<22} \x1b[90m{d:<8} \x1b[90m{s:<14} \x1b[33m{s:<22} \x1b[95m{s:<28} \x1b[90m{s:<10} {s:<10} {s:<10}\x1b[0m\n\n",
+            .{ self.name, self.total_operations, total_time_str, avg_std_str, min_max_str, p75_str, p99_str, p995_str },
+        );
+    }
+};
+
+pub fn prettyPrintHeader() !void {
+    const stdout = std.io.getStdOut().writer();
+    try stdout.print(
+        "\n{s:<22} {s:<8} {s:<14} {s:<22} {s:<28} {s:<10} {s:<10} {s:<10}\n",
+        .{ "benchmark", "runs", "total time", "time/run (avg ± σ)", "(min ... max)", "p75", "p99", "p995" },
+    );
+    try stdout.print("-----------------------------------------------------------------------------------------------------------------------------\n", .{});
+}
+
+pub const Percentiles = struct {
```

Comment:
Include a comment explaining how these percentiles are intended to be used or calculated, especially since percentiles are a key part of performance analysis.